rpcs3/Emu/Cell/Modules/cellAudioOut.cpp

```diff
@@ -177,6 +178,46 @@ audio_out_configuration::audio_out_configuration()
 	cellSysutil.notice("cellAudioOut: initial secondary output configuration: channels=%d, encoder=%d, downmixer=%d", secondary_output.channels, secondary_output.encoder, secondary_output.downmixer);
 }
 
+std::pair<AudioChannelCnt, AudioChannelCnt> audio_out_configuration::audio_out::get_channel_count_and_downmixer() const
```

Comment:
In the previous PR you said that the first element of the pair that this function returns is input channel count, but isn't input channel count specified when the game opens a port in cellAudio?
Over here:
https://github.com/RPCS3/rpcs3/blob/11c52306282fcbe3099805c59c1ea841208ed6a1/rpcs3/Emu/Cell/Modules/cellAudio.cpp#L1055
https://github.com/RPCS3/rpcs3/blob/11c52306282fcbe3099805c59c1ea841208ed6a1/rpcs3/Emu/Cell/Modules/cellAudio.h#L85-L91

Unless I'm missing something this doesn't need to return two values. You could probably just revert this commit 4beadb875363262388f4cdda0c92f285c27edcdf and that would fix all regressions, including the awful distortion.

Reply:
What does it have to do with this PR?

Reply:
Sorry, I just wanted to share my findings. I'll open an issue about it.

Reply:
Earthworm Jim HD

 RPCS3 v0.0.22-13796-ebabdd37 Alpha | master

 {PPU[0x1000000] Thread (main_thread) [HLE:0x00f506bc, LR:0x00153c84]} SIG: Thread terminated due to fatal error: Unsupported channel count for CELL_AUDIO_OUT_DOWNMIXER_TYPE_A in cellAudioOut config: 6
(in file C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\rpcs3\Emu\Cell\Modules\cellAudioOut.cpp:203[:24], in function get_channel_count_and_downmixer)

Reply:
That was caused by #12186

Reply:
@ capriots

Understood, just mentioning that this is still an issue - as this is supposed to be a follow-up to #12186